actions/websocket_actions.jsx

```diff
@@ -669,14 +669,19 @@ function handleDeleteTeamEvent(msg) {
 
 function handleUpdateMemberRoleEvent(msg) {
     const memberData = JSON.parse(msg.data.member);
+    const memberUserId = memberData.user_id;
+    const teamId = memberData.team_id;
+    const currentUserId = getCurrentUserId(getState());
     const newRoles = memberData.roles.split(' ');
 
-    dispatch(loadRolesIfNeeded(newRoles));
-
-    dispatch({
-        type: TeamTypes.RECEIVED_MY_TEAM_MEMBER,
-        data: memberData,
-    });
+    if (memberUserId === currentUserId) {
+        dispatch(loadRolesIfNeeded(newRoles));
+        dispatch({
+            type: TeamTypes.RECEIVED_MY_TEAM_MEMBER,
+            data: memberData,
+        });
+    }
+    dispatch(TeamActions.getTeamMember(teamId, memberUserId));
```

Comment:
Hinted at on the server PR, but this will have severe performance implications: the websocket event should contain everything the client needs and not require clients to reach back out to the server to fetch more information. In short, if there are 30,000 users online, this one websocket event will trigger 30,000 API requests at the same time, likely taking down the server.

Reply:
@lieut-data I see your point. I understand how this can have severe implications. Does it mean it doesn't make sense to fix this ticket for the sake of updating a modal?

Reply:
I can say with confidence that this ticket likely requires significantly more mana to achieve a secure and performant solution -- but I'll defer to @sbishel on the question of priority.

As to a more holistic solution, I'm still partial to the approach [discussed on community](https://community-release.mattermost.com/core/pl/dfomyzuj3frmtxd6drmf3m8syw):
> This might be an opportunity for UserB to somehow signal the server interest in this data, and temporarily subscribe to events as such. We don't have anything in the infrastructure to support that right now.

But I'm using the force here -- would need more investigation to determine feasibility.